lib/proxy/helpers.rb

```diff
@@ -83,6 +83,9 @@ def remote_fqdn(forward_verify = true)
     ip = request.env['REMOTE_ADDR']
     log_halt 403, 'could not get remote address from environment' if ip.empty?
 
+    # IPv6 with NIC identifier are not resolved
+    return ip if ip.sub(/%.*$/, '')
```

Comment:
This always resolves to true.
```suggestion
    return ip if ip.include?('%')
```